packages/salesforcedx-apex-debugger/src/adapter/apexDebug.ts

```diff
@@ -528,10 +528,10 @@ export class MapTupleContainer implements VariableContainer {
 
 export class ApexDebug extends LoggingDebugSession {
   protected myRequestService = new RequestService();
-  protected mySessionService: SessionService;
-  protected myBreakpointService: BreakpointService;
+  protected mySessionService!: SessionService;
+  protected myBreakpointService!: BreakpointService;
   protected myStreamingService = StreamingService.getInstance();
-  protected sfdxProject: string;
+  protected sfdxProject: string = '';
```

Comment:
Double-check that we do check that string == null or undefined to see if it is not initialized in the code. Just want to make sure that we don't assume an empty string means that it is set and ready to go.

Reply:
The usages of this do not seem to check if this is ever undefined.  ...which is kinda scary.  

ultimately, it ends up here:
```
const execution = new CliCommandExecutor(
      new SfdxCommandBuilder()
        .withArg('force:data:record:create')
        .withFlag('--sobjecttype', 'ApexDebuggerBreakpoint')
        .withFlag(
          '--values',
          `SessionId='${sessionId}' FileName='${typeref}' IsEnabled='true' Type='Exception'`
        )
        .withArg('--usetoolingapi')
        .withJson()
        .build(),
      { cwd: projectPath, env: this.requestService.getEnvVars() }
    ).execute();
```

Reply:
another option is for me to use ```string | undefined```.  

But to make Typescript happy when I pass it into functions that take ```string```,  I'll need to pass it like this ```this.sfdxProject || ''```, which brings back to the same issue.

From what I can tell there is no downstream undefined checks.

Reply:
This LGTM. It will either be the same valid value as `sfdxProject` in the user's launch.json or the default set in our package.json. The user can already set an empty string as their `sfdxProject` and it just falls back to the default.